src/chronify/time_configs.py

```diff
@@ -159,47 +158,137 @@ class AnnualTimeRange(TimeBaseModel):
     def list_time_columns(self) -> list[str]:
         return [self.time_column]
 
+    def list_time_zone_column(self) -> list[str]:
+        return []
 
-class IndexTimeRange(TimeBaseModel):
-    time_type: Literal[TimeType.INDEX] = TimeType.INDEX
-    start: int
+
+class IndexTimeRangeBase(TimeBaseModel):
+    """Defines a time range in the form of indexes"""
+
+    time_column: str = Field(description="Column in the table that represents index time.")
+    start: int = Field(description="starting index")
     length: int
-    resolution: timedelta
-    time_zone: TimeZone
+    start_timestamp: datetime = Field(
+        description="The timestamp represented by the starting index."
+    )
+    resolution: timedelta = Field(description="The resolution of time represented by the indexes.")
+    time_zone_column: Optional[str] = Field(
+        description="Column in the table that has time zone or offset information.", default=None
+    )
+    time_type: TimeType
+
+    @model_validator(mode="after")
+    def check_time_zone_column(self) -> "IndexTimeRangeBase":
+        tz_col = self.time_zone_column
+        if self.start_timestamp.tzinfo is not None and tz_col is not None:
+            msg = f"{self.start} is tz-aware and {tz_col=} is provided"
+            raise ConflictingInputsError(msg)
+        return self
+
+    def start_time_is_tz_naive(self) -> bool:
+        """Return True if the repreentative timestamps do not have time zones."""
+        return self.start_timestamp.tzinfo is None
 
     def list_time_columns(self) -> list[str]:
-        # TODO:
-        return []
+        return [self.time_column]
+
+    def list_time_zone_column(self) -> list[str]:
+        if self.time_zone_column is None:
+            return []
+        return [self.time_zone_column]
+
+
+class IndexTimeRangeNTZ(IndexTimeRangeBase):
+    """Index time that represents tz-naive timestamps.
+    start_timestamp is tz-naive
+    time_zone_column is None
+    """
+
+    time_type: Literal[TimeType.INDEX_NTZ] = TimeType.INDEX_NTZ
+    time_zone_column: None = None
+
+    @field_validator("start_timestamp")
+    @classmethod
+    def check_start_timestamp(cls, start_timestamp: datetime) -> datetime:
+        if start_timestamp.tzinfo is not None:
+            msg = "start_timestamp must be tz-naive for IndexTimeRangeNTZ model"
+            raise InvalidParameter(msg)
+        return start_timestamp
+
+
+class IndexTimeRangeTZ(IndexTimeRangeBase):
+    """Index time that represents tz-aware timestamps of a single time zone.
+    start_timestamp is tz-aware
+    time_zone_column is None
+    """
+
+    time_type: Literal[TimeType.INDEX_TZ] = TimeType.INDEX_TZ
+    time_zone_column: None = None
```

Comment:
Related to my other comment about time_zone_column: a user could pass a value here and that would likely break everything. You could write a validator to check that they don't. It would be better if the field did not exist.